experimental/Schemes/SpecOpen.jl

```diff
@@ -0,0 +1,675 @@
+export SpecOpen, ambient, gens, complement, npatches, affine_patches, intersections, name, intersect, issubset, closure, find_non_zero_divisor, is_non_zero_divisor, is_dense
+
+export StructureSheafRing, scheme, domain, OO
+
+export StructureSheafElem, domain, restrictions, patches, restrict, npatches
+
+export SpecOpenMor, maps_on_patches, restriction, identity_map, preimage, generic_fractions, pullback, maximal_extension
+
+@Markdown.doc """
+    SpecOpen{BRT, BRET, RT, RET, MST} <: Scheme{BRT, BRET}
+
+Zariski open subset ``U`` of an affine scheme ``X = Spec(R)``. 
+This stores a list of generators ``f₁,…,fᵣ`` of an ideal 
+``I`` defining the complement ``Z = X ∖ U``. 
+The scheme ``X`` is referred to as the *ambient scheme* and 
+the list ``f₁,…,fᵣ`` as the *generators* for ``U``.
+"""
+@attributes mutable struct SpecOpen{BRT, BRET, RT, RET, MST} <: Scheme{BRT, BRET}
+  X::Spec{BRT, BRET, RT, RET, MST} # the ambient scheme
+  gens::Vector{RET} # a list of functions defining the complement of the open subset
+
+  # fields used for caching
+  name::String
+  patches::Vector{Spec{BRT, BRET, RT, RET, MST}}
+  intersections::Dict{Tuple{Int}, Spec{BRT, BRET, RT, RET, MST}}
```

Comment:
```suggestion
  intersections::Dict{Int, Spec{BRT, BRET, RT, RET, MST}}
```
Do you really have one element tuples?

Reply:
No, not 1-element tuples, but tuples with entries of one and the same type, namely `Int`.

Reply:
Going to the REPL, I see the error now. Thanks.